src/ImageSharp/Formats/Jpeg/Components/Encoder/HuffmanScanEncoder.cs

```diff
@@ -388,5 +388,34 @@ private void FlushInternalBuffer()
                 this.target.Write(this.emitBuffer, 0, this.emitLen);
             }
         }
+
+        /// <summary>
+        /// Calculates how many minimum bits needed to store given value for Huffman jpeg encoding.
+        /// This method does not follow the standard convention - it does not support input value of zero.
+        /// </summary>
+        /// <remarks>
+        /// Passing zero as input value would result in an undefined behaviour.
+        /// This is done for performance reasons as Huffman encoding code checks for zero value, second identical check would degrade the performance in the hot path.
+        /// </remarks>
+        /// <param name="value">The value.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int GetHuffmanEncodingLegth(uint value)
+        {
+            DebugGuard.IsFalse(value == 0, nameof(value), "Huffman encoding does not encode zero values");
+#if SUPPORTS_BITOPERATIONS
+            // This should have been implemented as (BitOperations.Log2(value) + 1) as in non-intrinsic implementation
+            // But internal log2 is implementated like this: (31 - (int)Lzcnt.LeadingZeroCount(value))
+
+            // BitOperations.Log2 implementation also checks if input value is zero for the convention
+            // As this is a very specific method for a very specific Huffman encoding code
+            // We can omit zero check as this should not be invoked with value == 0 and guarded in debug builds
+            return 32 - System.Numerics.BitOperations.LeadingZeroCount(value);
+#else
+            // On the contrary to BitOperations implementation this does follow the convention and supports value == 0 case
+            // Although it's still won't be called with value == 0
+            // As these implementations behave differently for the value == 0 case it's documented as undefined behaviour
+            return Numerics.Log2SoftwareFallback(value) + 1;
```

Comment:
I don't think we should expose that as a separate method. 

I'd stick compiler conditionals in a `Numerics.Log2` method instead.

Reply:
I completely messed up with those comments as I completely forgot that huffman encoder can encounter a zero value during dc part encoding. Realistically, log2 is the solution for this method but `32 - lzcnt` is faster hardware-wise as there no if-check if lzcnt is supported. Or maybe I didn't understood correctly what you said about polyfill.

Reply:
@JimBobSquarePants sorry for namedropping, I'm not sure github sends notifications without tagging, at least I've seen your review by a chance.

You meant borrowing code from runtime source code? `BitOperations.LeadingZeroCount` is a polyfill itself, it has fallback code if hardware doesn't support `lzcnt`. It's really specific to jpeg right now plus fallback code assumes that input value is constrained in certain way, placing it in Numerics might not be good considering that.

Reply:
What I meant was implement a `Numerics.LeadingZeroCount(uint value)` method so that your calling code would be simplified.

Something like this. (_I think this is correct, patched from runtime source_).

```c#
public static int LeadingZeroCount(uint value)
{
#if SUPPORTS_BITOPERATIONS
    return BitOperations.LeadingZeroCount(value);
#else
    // Software fallback
    // Unguarded fallback contract is 0->31, BSR contract is 0->undefined
    if (value == 0)
    {
        return 32;
    }

    return 31 ^ Log2SoftwareFallback(value);
#endif
```

Reply:
Oh, now I got it, thanks!
That would lead to unnecesarry operations in jpeg hot path code - I've updated correct implementation, all tests passed (build failed on some strange decoding test). Worth doing only if it's needed somewhere else in the lib,

Reply:
Ah ok, so you're concerned about the branching in older target frameworks. Ok, make sense.